model/pod/podutil.go

```diff
@@ -0,0 +1,117 @@
+package pod
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/pkg/errors"
+)
+
+const (
+	executable = "./evergreen"
+	shell      = "CMD-SHELL "
+)
+
+// Constants representing default curl retry arguments.
+const (
+	curlDefaultNumRetries = 10
+	curlDefaultMaxSecs    = 100
+)
+
+// CurlCommandWithRetry returns the command to curl the evergreen client and retries the request.
+func (p *Pod) CurlCommandWithRetry(settings *evergreen.Settings, numRetries, maxRetrySecs int) (string, error) {
+	var retryArgs string
+	if numRetries != 0 && maxRetrySecs != 0 {
+		retryArgs = " " + curlRetryArgs(numRetries, maxRetrySecs)
+	}
+	cmds, err := p.curlCommands(settings, retryArgs)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	return strings.Join(cmds, " && "), nil
+}
+
+// CurlCommandWithDefaultRetry is the same as CurlCommandWithRetry using the
+// default retry parameters.
+func (p *Pod) CurlCommandWithDefaultRetry(settings *evergreen.Settings) (string, error) {
```

Comment:
There's no need to have two separate `CurlCommandWithRetry` and `CurlCommandWithDefaultRetry` methods. You can just make one `CurlCommand` method that uses the default retry values.